fleetspeak/src/client/config/filesystem_persistence_handler.go

```diff
@@ -40,63 +41,67 @@ type FilesystemPersistenceHandler struct {
 // files. Possible files include:
 //
 // /communicator.txt    - A text format clpb.CommunicatorConfig, used to tweak communicator behavior.
-// /writeback           - Used to maintain state across restarts,
 // /services/<service>  - A binary format SignedClientServiceConfig. One file for each configured service.
 //
 // All of these files are optional, though Fleetspeak will not be particularly
 // useful without at least one configured service.
 //
-// If readonly is true, the client will not attempt to write to
-// <ConfigurationPath>/writeback, in order to preserve client identity.
-//
-// readonly is intended for testing and specialized applications - should be
-// hardcoded false in normal deployments.
-func NewFilesystemPersistenceHandler(configurationPath string, readonly bool) (*FilesystemPersistenceHandler, error) {
+// If stateFile is nonempty, the client will attempt to write persistent state
+// to it, in order to preserve identity across restarts.
+func NewFilesystemPersistenceHandler(configurationPath, stateFile string) (*FilesystemPersistenceHandler, error) {
 	if err := verifyDirectoryPath(configurationPath); err != nil {
 		return nil, fmt.Errorf("invalid configuration path: %v", err)
 	}
+	if stateFile != "" {
+		if err := verifyDirectoryPath(path.Dir(stateFile)); err != nil {
+			log.Warningf("Could not state statefile location [%v], continuing as an ephemeral client: %v", path.Dir(stateFile), err)
```

Comment:
This fallback seems really dangerous, if you have a config bug you will spam the db with tons of new client ids. Better to error out here?

Reply:
Done.